core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1693,18 +1694,24 @@ public Status onResult2(final ResolutionResult resolutionResult) {
         return Status.OK;
       }
 
-      List<EquivalentAddressGroup> servers = resolutionResult.getAddresses();
-      channelLogger.log(
-          ChannelLogLevel.DEBUG,
-          "Resolved address: {0}, config={1}",
-          servers,
-          resolutionResult.getAttributes());
-
-      if (lastResolutionState != ResolutionState.SUCCESS) {
-        channelLogger.log(ChannelLogLevel.INFO, "Address resolved: {0}", servers);
-        lastResolutionState = ResolutionState.SUCCESS;
+      StatusOr<List<EquivalentAddressGroup>> serversOrError =
+          resolutionResult.getAddressesOrError();
+      if (serversOrError != null && serversOrError.hasValue()) {
+        channelLogger.log(
+            ChannelLogLevel.DEBUG,
+            "Resolved address: {0}, config={1}",
+            serversOrError,
+            resolutionResult.getAttributes());
+
+        if (lastResolutionState != ResolutionState.SUCCESS) {
+          channelLogger.log(ChannelLogLevel.INFO, "Address resolved: {0}",
+              serversOrError.getValue());
+          lastResolutionState = ResolutionState.SUCCESS;
+        }
+      } else if (serversOrError != null && !serversOrError.hasValue()) {
```

Comment:
Since this block returns, you may consider making it separate early return (not if/else):
```java
if (!serversOrError.hasValue()) {
  handleErrorInSyncContext(serversOrError.getStatus());
  return serversOrError.getStatus();
}

List<EquivalentAddressGroup> servers = serversOrError.getValue();
// the rest of the previous code
```

Reply:
Done.